email_mailbox/src/utils/electronEventInterface.js

```diff
@@ -154,33 +175,37 @@ const parseAndStoreEventsBatch = async ({ events, hasMoreEvents }) => {
         sendFetchEmailsErrorMessage();
       }
     }
+    i++;
   }
 
   const rowIdsFiltered = rowIds.filter(rowId => !!rowId);
-  if (rowIdsFiltered.length || (events.length && !rowIdsFiltered.length)) {
+  if (rowIdsFiltered.length || (eventsSize && !rowIdsFiltered.length)) {
     if (rowIdsFiltered.length) await setEventAsHandled(rowIdsFiltered);
 
-    const labelIds = labelIdsEvent.size ? Array.from(labelIdsEvent) : null;
-    const threadIds = threadIdsEvent.size ? Array.from(threadIdsEvent) : null;
-    const badgeLabelIds = badgeLabelIdsEvent.size
-      ? Array.from(badgeLabelIdsEvent)
-      : null;
-    const labels = labelsEvent
-      ? Object.keys(labelsEvent).length
-        ? labelsEvent
-        : null
-      : null;
-    const hasStopLoad = !hasMoreEvents;
-    emitter.emit(Event.STORE_LOAD, {
-      avatarHasChanged,
-      completedTask: totalEmailsHandled,
-      labelIds,
-      threadIds,
-      labels,
-      badgeLabelIds,
-      hasStopLoad
-    });
+    if (!shouldBreakEvents) {
+      const labelIds = labelIdsEvent.size ? Array.from(labelIdsEvent) : null;
+      const threadIds = threadIdsEvent.size ? Array.from(threadIdsEvent) : null;
+      const badgeLabelIds = badgeLabelIdsEvent.size
+        ? Array.from(badgeLabelIdsEvent)
+        : null;
+      const labels = labelsEvent
+        ? Object.keys(labelsEvent).length
+          ? labelsEvent
+          : null
+        : null;
+      const hasStopLoad = !hasMoreEvents;
+      emitter.emit(Event.STORE_LOAD, {
+        avatarHasChanged,
+        completedTask: totalEmailsHandled,
+        labelIds,
+        threadIds,
+        labels,
+        badgeLabelIds,
+        hasStopLoad
+      });
+    }
   }
+  return;
```

Comment:
I think you don't need to return at the end of a function

Reply:
Yes, I need to know when this badge finished... a wait this function to end